src/frontend/src/lib/components/core/Menu.svelte

```diff
@@ -19,37 +20,57 @@
 	import { OISY_REPO_URL } from '$lib/constants/oisy.constants';
 	import { AppPath } from '$lib/constants/routes.constants';
 	import { NAVIGATION_MENU_BUTTON, NAVIGATION_MENU } from '$lib/constants/test-ids.constants';
+	import { networkId } from '$lib/derived/network.derived';
 	import { i18n } from '$lib/stores/i18n.store';
 	import {
 		isRouteActivity,
 		isRouteDappExplorer,
 		isRouteSettings,
-		isRouteTokens
+		isRouteTokens,
+		isRouteTransactions,
+		networkUrl
 	} from '$lib/utils/nav.utils';
 
 	let visible = false;
 	let button: HTMLButtonElement | undefined;
 
+	let fromRoute: NavigationTarget | null;
+
+	afterNavigate(({ from }) => {
+		fromRoute = from;
+	});
+
+	let isTransactionsRoute = false;
+	$: isTransactionsRoute = isRouteTransactions($page);
+
 	const hidePopover = () => (visible = false);
 
 	const goToTokens = async () => {
 		hidePopover();
-		await goto(AppPath.Tokens);
+		await goto(
+			networkUrl({ path: AppPath.Tokens, networkId: $networkId, isTransactionsRoute, fromRoute })
```

Comment:
Thinking at it now, maybe the parameter `isTransactionsRoute` should not be called like that but be more generic like "resetNetwork` or something. Not related to this PR, just thinking out at loud.

Reply:
good point! it should indicate that "it should use the network id that was in the page before"